app/src/androidTest/java/sw19/moring03/paint/LineTypeTouchTest.java

```diff
@@ -0,0 +1,44 @@
+package sw19.moring03.paint;
+
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+@RunWith(AndroidJUnit4.class)
+public class LineTypeTouchTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> launchActivityRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Test
+    public void clickOnDashedLine() {
+        onView(withId(R.id.lineTypeChooserButton)).perform(click());
+        onView(withId(R.id.dashedLine)).perform(click());
```

Comment:
Can you please check here that having clicked on the buttons had the desired effect (correct lineEffect has been set)? So far these tests only perform actions but don't check whether they've been successful.